components/ws-manager/pkg/manager/status.go

```diff
@@ -605,7 +607,7 @@ func extractFailure(wso workspaceObjects) (string, *api.WorkspacePhase) {
 	}
 
 	status := pod.Status
-	if status.Phase == corev1.PodFailed {
+	if status.Phase == corev1.PodFailed && (status.Reason != "" || status.Message != "") {
```

Comment:
is it guaranteed that there is non-empty `status` in case a pod failed?

Reply:
`pod.Status` is a struct (not a pointer) to a struct, hence cannot be `nil`, if that's what you mean.
If the `status` is empty (in the sense of Go's default value), `status.Phase == ""` and we won't enter here.

Most of those conditions - in fact the whole status machinery - is derived purely from observing system behaviour. Until recently we hadn't seen pod failure with an empty `reason` and/or `message`. Now that we do, we add a condition and test :)

Reply:
> Until recently we hadn't seen pod failure with an empty reason and/or message. Now that we do, we add a condition and test :)

that's great!